src/onedrive.rs

```diff
@@ -269,56 +270,83 @@ impl OneDrive {
             .await
     }
 
-    /// Create a new folder under an `DriveItem`
-    ///
-    /// Create a new folder [`DriveItem`][drive_item] with a specified parent item or path.
+    /// Creates a new [`DriveItem`][drive_item] allowing to set all supported attributes.
     ///
     /// # Errors
     /// Will result in `Err` with HTTP `409 CONFLICT` if [`conflict_behavior`][conflict_behavior]
     /// is set to [`Fail`][conflict_fail] and the target already exists.
     ///
     /// # See also
-    /// [Microsoft Docs](https://docs.microsoft.com/en-us/graph/api/driveitem-post-children?view=graph-rest-1.0)
     ///
+    /// [Microsoft Docs](https://learn.microsoft.com/en-us/graph/api/resources/driveitem?view=graph-rest-1.0)
+    ///
+    /// [with_opt]: #method.create_folder_with_option
     /// [drive_item]: ./resource/struct.DriveItem.html
     /// [conflict_behavior]: ./option/struct.DriveItemPutOption.html#method.conflict_behavior
     /// [conflict_fail]: ./enum.ConflictBehavior.html#variant.Fail
-    pub async fn create_folder_with_option<'a>(
+    pub async fn create_drive_item<'a>(
         &self,
         parent_item: impl Into<ItemLocation<'a>>,
-        name: &FileName,
-        option: DriveItemPutOption,
+        drive_item: DriveItem,
+        conflict_behavior: Option<ConflictBehavior>,
```

Comment:
I  noticed that currently `create_folders*` uses `DriveItemPutOption` which includes additional `If-{Non-,}Match` support. If that's valid, we should go with:

```suggestion
        option: DriveItemPutOption,
```
and `.apply(option)` on the request builder.

Reply:
Done